js_modules/dagster-ui/packages/ui-core/src/overview/OverviewSchedules.tsx

```diff
@@ -56,16 +62,25 @@ export const OverviewSchedules = () => {
     },
   );
   const {data, loading} = queryResultOverview;
-  useBlockTraceOnQueryResult(queryResultOverview, 'OverviewSchedulesQuery');
+  useBlockTraceUntilTrue('OverviewSchedules', !!data || !workspaceLoading);
 
   const refreshState = useQueryRefreshAtInterval(queryResultOverview, FIFTEEN_SECONDS);
 
   const repoBuckets = useMemo(() => {
     const visibleKeys = visibleRepoKeys(visibleRepos);
-    return buildBuckets(data).filter(({repoAddress}) =>
+    const cachedEntries = Object.values(cachedData).filter(
+      (location): location is Extract<typeof location, {__typename: 'WorkspaceLocationEntry'}> =>
+        location.__typename === 'WorkspaceLocationEntry',
+    );
+    const workspaceOrError = data?.workspaceOrError;
+    const entries =
+      workspaceOrError?.__typename === 'Workspace'
+        ? workspaceOrError.locationEntries
+        : cachedEntries;
```

Comment:
These ~8 lines seems to be duplicated 4x, maybe we break it out into a named helper method? I think giving this bit a name might be nice to clarify the cache / fallback behavior implemented here.